### PR TITLE
Fix for Pystac ItemCollections

### PR DESCRIPTION
**Pull request by scottyhq**
Follow on to #64 to work with Pystac ItemCollection https://pystac.readthedocs.io/en/latest/api.html#itemcollection. 

Partially addresses #65, but did not update all the docs and examples

Not sure if I should add the poetry.lock file that I updated to test this out locally

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -6,7 +6,7 @@ license = "MIT"
 name = "stackstac"
 readme = "README.md"
 repository = "https://github.com/gjoseph92/stackstac"
-version = "0.2.1"
+version = "0.2.2"
```
Comment (gjoseph92) [inline pyproject.toml:9]:
```suggestion
version = "0.2.1"
```
I'd prefer to bump the stackstac version as a separate commit if that's okay?

File: pyproject.toml [hunk 2]
```diff
@@ -54,7 +54,7 @@ graphviz = "^0.16"
 jupyterlab = "^3.0.14"
 keyring = "^23.0.1"
 py-spy = "^0.3.4"
-pystac = "^0"
+pystac = "^1"
```
Comment (gjoseph92) [inline pyproject.toml:57]: Should we be any stricter on minor versions? Or is any 1.x fine? Probably is fine, I'm just checking.

**Review by gjoseph92 — COMMENTED** [review]
Thanks for the fix @scottyhq and thanks for updating docs as well!

If you want to add some tests, by all means go for it. My time to work on stackstac is very limited right now, so I'm sure I won't get to it for a while. 

> Not sure if I should add the poetry.lock file that I updated to test this out locally

Yes, please add this (see https://python-poetry.org/docs/basic-usage/#commit-your-poetrylock-file-to-version-control). That'll be especially helpful once we have CI.

File: pyproject.toml (hunk 2 above)
Comment (scottyhq) [reply pyproject.toml:57]: less restrictive seems best, especially with pystac as a 'dev' dependency (although might be worth getting more specific if it becomes a core dependency #20)? If I understand correctly these dev pins are only considered when running `poetry update` where the lockfile is solved for?

Comment (gjoseph92) [reply pyproject.toml:57]: Yes I think that's right. `poetry update` gives you the latest versions of everything within the bounds specified in the lockfile. Agreed that we should leave it like this for now, and we'll get more restrictive if necessary in the future.

**Comment by lucadelu** [comment]
Hi, 
sorry I tried to install from master using `pip install git+https://github.com/gjoseph92/stackstac` but I continue to have this problem

```
TypeError                                 Traceback (most recent call last)
<ipython-input-26-d03ca319dd0a> in <module>
----> 1 ds = stackstac.stack(planetary_search.get_all_items())

~/gitfem/euromammals/venv/lib/python3.9/site-packages/stackstac/stack.py in stack(items, assets, epsg, resolution, bounds, bounds_latlon, snap_bounds, resampling, chunksize, dtype, fill_value, rescale, sortby_date, xy_coords, properties, band_coords, gdal_env, errors_as_nodata, reader)
    278         automatically computed from the items you pass in.
    279     """
--> 280     plain_items = items_to_plain(items)
    281 
    282     if sortby_date is not False:

~/gitfem/euromammals/venv/lib/python3.9/site-packages/stackstac/stac_types.py in items_to_plain(items)
    157         return [item.to_dict() for item in items.features]
    158 
--> 159     raise TypeError(f"Unrecognized STAC collection type {type(items)}: {items!r}")

TypeError: Unrecognized STAC collection type <class 'pystac.item_collection.ItemCollection'>: <pystac.item_collection.ItemCollection object at 0x7f420405b640>
```

My code is

```
planetary_catalog = pystac_client.Client.open(
    "https://planetarycomputer.microsoft.com/api/stac/v1"
)
planetary_search = planetary_catalog.search(
    collections=["sentinel-2-l2a"],
    bbox=bbox.values[0],
    datetime=date_range,
)
ds = stackstac.stack(planetary_search.get_all_items())
```

I'm doing something wrong?

```
print(pystac.version.get_stac_version())
print(pystac_client.__version__)
1.0.0
0.3.0
```

**Comment by gjoseph92** [comment]
Try `pip install --upgrade git+https://github.com/gjoseph92/stackstac`? What you're doing works for me on `main`.

Sorry I haven't gotten a release out. I was testing things out for a release and noticed some intermittent failures when upgrading rasterio (which feel thread-safety related), so I wanted to verify that first in case we need to pin versions for now.

**Comment by lucadelu** [comment]
In my case at the end I had to remove completely stackstac and reinstall it. Thanks a lot

**Comment by gjoseph92** [comment]
Hey @amfriesz and @lucadelu, I finally got a release of stackstac out. Try updating from pip now!

**Pull request merged**